### PR TITLE
Make worker docker volume size configurable via CR

### PR DESCRIPTION
**Pull request by tuommaki**
Same docker volume size doesn't fit to all tenant cluster use cases. It
makes sense to be able to configure it per guest cluster. Earlier
implementation always defaulted to 100GB (which is the default now as
well). When the field is present, its value is used instead of default
one.

### CODE REVIEW

File: service/controller/v16/key/key.go [hunk 1]
```diff
@@ -506,6 +506,28 @@ func WorkerCount(customObject v1alpha1.AWSConfig) int {
...
+// size.
+func WorkerDockerVolumeSizeGB(customObject v1alpha1.AWSConfig) (int, error) {
+	if len(customObject.Spec.AWS.Workers) <= 0 {
+		return 0, nil
+	}
+
+	v := customObject.Spec.AWS.Workers[0].DockerVolumeSizeGB
+
+	if v == "" {
+		return 0, nil
```
Comment (kopiczko) [inline service/controller/v16/key/key.go]: Should it be 0 or 100?

File: service/controller/v16/resource/cloudformation/current.go [hunk 2]
```diff
@@ -99,6 +100,32 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
+			if cloudformationservice.IsOutputNotFound(err) {
+				// Since we are transitioning between versions we will have situations in
+				// which old clusters are updated to new versions and miss the docker
+				// volume resource name in the CF stack outputs. We ignore this problem
+				// for now and move on regardless. On the next resync period the output
+				// value will be there, once the cluster got updated.
+				//
+				// TODO remove this condition as soon as all guest clusters in existence
+				// obtain a docker volume size.
+				dockerVolumeSizeGB = "100"
```
Comment (kopiczko) [inline service/controller/v16/resource/cloudformation/current.go]: Why this is set here? Where is it declared or used?

File: service/controller/v16/key/key.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/v16/key/key.go]: It looks counterintuitive to me. Is there a reason why it's like that? Can that be changed so we return the default here?

Comment (xh3b4sd) [reply service/controller/v16/key/key.go]: +1 on defaulting in the key package. We do similar things here already.

File: service/controller/v16/resource/cloudformation/current.go [hunk 3]
```diff
@@ -99,6 +100,32 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
+			if cloudformationservice.IsOutputNotFound(err) {
+				// Since we are transitioning between versions we will have situations in
+				// which old clusters are updated to new versions and miss the docker
+				// volume resource name in the CF stack outputs. We ignore this problem
+				// for now and move on regardless. On the next resync period the output
+				// value will be there, once the cluster got updated.
+				//
+				// TODO remove this condition as soon as all guest clusters in existence
+				// obtain a docker volume size.
+				v = "100"
```
Comment (xh3b4sd) [inline service/controller/v16/resource/cloudformation/current.go]: Lets please have a TODO issue for this.

Comment (tuommaki) [reply service/controller/v16/resource/cloudformation/current.go]: https://github.com/giantswarm/giantswarm/issues/4139

**Pull request merged**